Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -0,0 +1,2212 @@
+                               std::vector<double> &vec_xmax) {
+  // Validate
+  if (vec_xmin.size() != vec_xmax.size())
+    throw runtime_error("Sizes of xmin and xmax (vectors) are not equal. ");
+
+  // Note: after testing it is found that multi-domain Fit cannot be reused
+  API::IAlgorithm_sptr fit;
+  try {
+    fit = createChildAlgorithm("Fit", -1, -1, false);
+  } catch (Exception::NotFoundError &) {
+    std::stringstream errss;
+    errss << "The FitPeak algorithm requires the CurveFitting library";
+    throw std::runtime_error(errss.str());
+  }
+  // set up background fit instance
+  fit->setProperty("Minimizer", m_minimizer);
+  fit->setProperty("CostFunction", m_costFunction);
+  fit->setProperty("CalcErrors", true);
+
+  // This use multi-domain; but does not know how to set up IFunction_sptr
+  // fitfunc,
+  boost::shared_ptr<MultiDomainFunction> md_function =
+      boost::make_shared<MultiDomainFunction>();
+
+  // Set function first
+  md_function->addFunction(fit_function);
+
+  //  set domain for function with index 0 covering both sides
+  md_function->clearDomainIndices();
+  std::vector<size_t> ii(2);
+  ii[0] = 0;
+  ii[1] = 1;
+  md_function->setDomainIndices(0, ii);
+
+  // Set the properties
+  fit->setProperty("Function",
+                   boost::dynamic_pointer_cast<IFunction>(md_function));
+  fit->setProperty("InputWorkspace", dataws);
+  fit->setProperty("WorkspaceIndex", static_cast<int>(wsindex));
+  fit->setProperty("StartX", vec_xmin[0]);
+  fit->setProperty("EndX", vec_xmax[0]);
+  fit->setProperty("InputWorkspace_1", dataws);
+  fit->setProperty("WorkspaceIndex_1", static_cast<int>(wsindex));
+  fit->setProperty("StartX_1", vec_xmin[1]);
+  fit->setProperty("EndX_1", vec_xmax[1]);
+  fit->setProperty("MaxIterations", 50);
+
+  // Execute
+  fit->execute();
+  if (!fit->isExecuted()) {
+    throw runtime_error("Fit is not executed on multi-domain function/data. ");
+  }
+
+  // Retrieve result
+  std::string fitStatus = fit->getProperty("OutputStatus");
+  //  m_sstream << "[DB] Multi-domain fit status: " << fitStatus << ".\n";
+
+  double chi2 = DBL_MAX;
+  if (fitStatus == "success") {
+    chi2 = fit->getProperty("OutputChi2overDoF");
+  }
+
+  return chi2;
+}
+
+//----------------------------------------------------------------------------------------------
+/** Fit peak with high background
+ * @brief FitPeaks::FitFunctionHighBackground
+ * @param fit
+ * @param fit_window
+ * @param ws_index
+ * @param expected_peak_center
+ * @param observe_peak_width
+ * @param peakfunction
+ * @param bkgdfunc
+ * @param high_bkgd_function
+ * @return
+ */
+double FitPeaks::FitFunctionHighBackground(
+    IAlgorithm_sptr fit, const std::pair<double, double> &fit_window,
+    const size_t &ws_index, const double &expected_peak_center,
+    bool observe_peak_width, API::IPeakFunction_sptr peakfunction,
+    API::IBackgroundFunction_sptr bkgdfunc,
+    API::IBackgroundFunction_sptr high_bkgd_function) {
+  // F
```

Comment:
`FitPeaks.cpp:2197:16: warning: do not use 'compare' to test equality of strings; use the string equality operator instead [misc-string-compare]`